### ISSUE TITLE
Support for general function/Hilbert spaces

### ISSUE THREAD

**Issue by kiranvad** [geomstats/geomstats#1194]
Hello everyone,
Really enjoy this package both for understanding and computing with differential geometry and manifolds.
This is a follow-up discussion with @ninamiolane on Twitter about the prospect of having general function spaces or Hilbert spaces as manifolds along with the discrete curves.
Lots of experimental data collected in material science is in the form of functions (optical spectroscopy wavelength vs intensity). While this can in principle is considered as discrete curves in R2, the resulting metric would be suboptimal.
We recently started using functional spaces and their geometry to compute statistics and distances for this type of data and were wondering if generalizing them would be of interest to geomstats team.
I have a very crude implementation [here](https://github.com/kiranvad/geomstats/blob/948b6fb4650ff3e62ee63aaad94f5926f84a3b3d/geomstats/geometry/functions.py) for a few of the spaces such as the L2, infinite-dimensional Hypersphere, SRSF (which I think is very closely related to SRVF implemented herein).
Would love to chat more about this if there's sufficient interest either here or over email.

Thank you!
Kiran Vaddi

**Comment by ninamiolane** [comment]
Hi @kiranvad thank you very much for reaching out! Good to have you here.

Indeed, geomstats implementation only focuses on [discretized curves](https://github.com/geomstats/geomstats/blob/master/geomstats/geometry/discrete_curves.py) with a fixed number of sampling points. [See a notebook on this here](https://github.com/geomstats/geomstats/blob/master/notebooks/08_shape_analysis.ipynb).

Your implementation would generalize this?

Additionally, I see that you are writing methods related to information geometry: fantastic! Do you think these methods could be integrated into our model [information_geometry](https://github.com/geomstats/geomstats/tree/master/geomstats/information_geometry)? S[ee a notebook on this here](https://github.com/geomstats/geomstats/blob/master/notebooks/06_information_geometry.ipynb).

To summarize, there is definitely a huge interest on our side to integrate your methods here! I would be happy to assist you in the integration of this code, and I believe that @alebrigant and @nguigs could be interested too.

**Comment by kiranvad** [comment]
That sounds great! 
The implementations I linked are pretty rudimentary but yes they are somewhat generalizations of discretized curves (cf.  [this book](https://link.springer.com/book/10.1007/978-1-4939-4020-2)) and I would love to work with all of you to improve them and learn more.
I will try to put together a pull request for these and maybe we can go from there?

**Comment by nguigs** [comment]
Hi @kiranvad, thanks for your interest and your first implementation! There indeed seems to be many things we could integrate into geomstats! Could you split your contribution into small PRs for each new manifold/metric pair so that we can discuss them in detail? Looking forward to it!
